package/gargoyle/files/www/js/common.js

```diff
@@ -1283,6 +1328,68 @@ function setAllowableSelections(selectId, allowableValues, allowableNames, contr
 	}
 }
 
+// Add hidden option group. Adding an option to this group will unhide it.
+function addOptGroup(select, label, hidden = true, disabled = false)
+{
+	const optgroup = document.createElement("optgroup");
+	optgroup.label = label;
+	optgroup.hidden = hidden;
+	optgroup.disabled = disabled;
+	select.appendChild(optgroup);
+	return optgroup;
+}
+
+// Add option and unhide its group if present.
+function addOption(selectOrOptgroup, text, value = "", selected = false, hidden = false, disabled = false)
```

Comment:
We already have addOptionToSelectElement. Can these be consolidated while maintaining compatibility with existing code? No sense having two. If not, this function should be made specifically to add to OptGroups only (and documented as such) or moved to qr_code_commn.js.

Reply:
I think it's currently a bit too different from existing functions (DOM object vs ID string as arguments) and unless used elsewhere it's better too keep `common.js` small.